### PR TITLE
Support key-per-participant e2ee

### LINKED ISSUE #1278: E2EE first cut: Generate media key, send an encrypted room event on call join
### Your use case

This bug covers:
 * [x] Generating some random bytes to use for an E2EE media key
 * [x] Implementing a custom E2EEKeyProvider for LiveKit to provide per-participant keys
 * [x] Sending our key to the room as a room event when we join the call
 * [x] Reading media keys from room events when we join the call and adding them to our key provider
 * [x] Reading new events as new members join the call and adding them to our key provider

Things we removed from this bug:
 * Checking whether LiveKit will tell us when we're getting encrypted media or whether we need to signal it OOB: I suspect the latter because if the browser can't do E2EE then it won't b able to inspect the packets to discover that they're scrambled, in which case we should add a field to the call event (I think?) to announce the fact it's intended to be e2ee.
 * Reading the field of the state event and enabling / disabling e2ee accordingly
   * We used the absence of a canonical alias as the indicator of whether a room is e2ee or not instead of doing these

This gets a very basic first version of end to end encryption with automatic key exchange

### Have you considered any alternatives?

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by dbkr**
Replaces https://github.com/vector-im/element-call/pull/1350
Requires https://github.com/matrix-org/matrix-js-sdk/pull/3667
Fixes https://github.com/vector-im/element-call/issues/1278

### CODE REVIEW

File: src/widget.ts [hunk 1]
```diff
@@ -77,6 +77,8 @@ export const widget = ((): WidgetHelpers | null => {
       logger.info("Widget API is available");
       const api = new WidgetApi(widgetId, parentOrigin);
       api.requestCapability(MatrixCapabilities.AlwaysOnScreen);
+      api.requestCapabilityToSendEvent(EventType.CallEncryptionKeysPrefix);
+      api.requestCapabilityToReceiveEvent(EventType.CallEncryptionKeysPrefix);
```
Comment (robintown) [inline src/widget.ts:81]: Is there a matching PR yet for matrix-react-sdk to auto-approve these capabilities?

File: src/livekit/useLiveKit.ts [hunk 2]
```diff
@@ -39,9 +40,16 @@ import {
   ECConnectionState,
   useECConnectionState,
 } from "./useECConnectionState";
+import { MatrixKeyProvider } from "../e2ee/matrixKeyProvider";
+
+export enum E2EEMode {
```
Comment (robintown) [inline src/livekit/useLiveKit.ts]: I'm confused why both this and `E2eeType` need to exist. Can we consolidate them?

**Review by robintown — APPROVED** [review]
Generally looks good to go

File: src/widget.ts (hunk 1 above)
Comment (dbkr) [reply src/widget.ts:81]: Yep, https://github.com/matrix-org/matrix-react-sdk/pull/11783 (and another previously from Simon adding them)

File: src/livekit/useLiveKit.ts (hunk 2 above)
Comment (dbkr) [reply src/livekit/useLiveKit.ts]: Hah, probably because I added one forgetting the other existed 😂

**Pull request merged**